More planning scripts

rubyvanrooyen opened the pull request:
Last two helper scripts to assist users offline. Both use astropy with the respective functionalities:
* `astrokat-coords.py` is a simple command line tool to convert Galactic coordinates (gal_lat, gal_lon) to equatorial coordinates (ra, dec)
* `astrokat-uvcoverage.py` give some indication of UV coverage dependent on array antennas selected

ajoubertza commented on scripts/astrokat-coords.py:
```diff
@@ -0,0 +1,194 @@
+if __name__ == '__main__':
+    # This code will take care of negative values for declination
+    for i, arg in enumerate(sys.argv):
+        if (arg[0] == '-') and arg[1].isdigit():
```
May also raise an exception if len(arg) < 2.

rubyvanrooyen replied:
Very good suggestion, much appreciated
I updated all the scripts that use this little hack to add this extra test

rubyvanrooyen commented on config/mkat_antennas.yml:
```diff
@@ -0,0 +1,69 @@
+reference:
+  latitude: '-30:42:47.4'
+  longitude: '21:26:38.0'
+  altitude: 1060.0
```
I used values from original ground measurements.
I updated the values to those in the config file

rubyvanrooyen commented on config/mkat_antennas.yml:
```diff
@@ -0,0 +1,69 @@
+  longitude: '21:26:38.0'
+  altitude: 1060.0
+antennas:
+  - name=m000, diameter=13.5, north=-8.21, east=26.797, up=0.117
```
I updated the values to be the most recent as per the delay config files.
I also added a helper script for doing the update in future, since the delay values are continually updating.

rubyvanrooyen commented:
@ajoubertza , @xinyuwu 

I updated the UV coverage script:
* cleaned up all the trailing units from astropy that caused unexpected errors
* separated the main simulation functionality to take specific parameters -- this should make it easier for you to call the main function from another script
* it now uses a target coordinate and observation duration to calculate the hour angle range

rubyvanrooyen commented:
@ajoubertza can you please indicate if I can merge this branch with master

ajoubertza commented on config/update_mkat_antennas.py:
```diff
@@ -0,0 +1,103 @@
+"""Update MeerKAT antenna position file"""
```
This is handy for future updates 👍

ajoubertza commented on scripts/astrokat-uvcoverage.py:
```diff
@@ -1,13 +1,16 @@
 
 import argparse
 import astropy.constants as phys
+import ephem
```
OK, but I thought this script was trying to do everything with astropy?

rubyvanrooyen replied:
astropy does not have an "observer" for that you have to use astroplan, which in itself is a very good tool, but it is >py3.5, so I had to use ephem for the interim until 2.7 is fully retired
Since ephem is only used for the observer, but the targets are astropy, I will sidestep the rounding issues of ephem in pointing

ajoubertza commented on scripts/astrokat-uvcoverage.py:
```diff
@@ -31,13 +34,15 @@ def __init__(self,
-        arr_desc = {'names': ('name', 'north', 'east', 'up', 'location'),
+        self.observer = None
+        arr_desc = {'names': ('name', 'east', 'north', 'up', 'location'),
                     'formats': ('S5', float, float, float, EarthLocation)}
```
Does S5 mean up to 5 chars?  I'm wondering about SKA / MKAT+ dishes, which will have names like `s0013`, in which case 5 is enough.

rubyvanrooyen replied:
Yes, s5 means max of 5 characters are read, so if you give more than 5 it truncates.
I used the meerkat array naming convention -- 5 should be sufficient.
